src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.uuuu");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
```

Comment:
This method is very complicated, I suggest you to break it down into separate methods. Create method that makes report, calculates sallary data and call them in getSalaryInfo method. As a result code will be more readable and maintainable

Reply:
Separate method for report seems unreasonable since it is created through entire length of method in different parts